sql/sql_table.cc

```diff
@@ -11460,16 +11460,11 @@ bool check_engine(THD *thd, const char *db_name,
   if (create_info->tmp_table() &&
       ha_check_storage_engine_flag(*new_engine, HTON_TEMPORARY_NOT_SUPPORTED))
   {
-    if (create_info->used_fields & HA_CREATE_USED_ENGINE)
-    {
-      my_error(ER_ILLEGAL_HA_CREATE_OPTION, MYF(0),
-               hton_name(*new_engine)->str, "TEMPORARY");
-      *new_engine= 0;
-      DBUG_RETURN(true);
-    }
-    *new_engine= myisam_hton;
+    my_error(ER_ILLEGAL_HA_CREATE_OPTION, MYF(0),
+             hton_name(*new_engine)->str, "TEMPORARY");
+    *new_engine= 0;
+    DBUG_RETURN(true);
```

Comment:
Let's document this behavior change properly. Like we discussed in the KB.

Reply:
Will do in https://mariadb.com/kb/en/create-table/#create-temporary-table